### PR TITLE
to cocina mapping:  originInfo eventType, dateOther

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made?

to address the problem described in #1426 (though based on validation against the fedora cache, this may also introduce a regression in the mapping, i'm not quite sure, so leaving in draft for pairing/discussion)

## How was this change tested?

* unit tests
* validation against the fedora cache

it seems like the change in this PR fixed most but not all of the `missing eventType` error (5417 down to 1341):
```
[deploy@sdr-deploy dor-services-app]$ 
[deploy@sdr-deploy dor-services-app]$ diff results-master-all_2020-11-13.txt results-dateOther-event-type-all_2020-11-13.txt
2c2
< Data error: 95317 of 2685579 (3.549216016359973%)
---
> Data error: 91316 of 2685579 (3.4002351075876005%)
79324,84741d79323
< Data error: [DATA ERROR] originInfo/dateOther missing eventType (5417 errors)
< druid:xv158sd4671
...snip 5414 druids...
< druid:zc893yf2565
< druid:ds794zt2803
88276c82858
< Data error: [DATA ERROR] Contributor type incorrectly capitalized (2667 errors)
---
> Data error: [DATA ERROR] Contributor type incorrectly capitalized (2742 errors)
88283a82866
> druid:mh797tm0200
...snip 73 druids...
> druid:qh821qx7318
93293a87951,89292
> Data error: [DATA ERROR] originInfo/dateOther missing eventType (1341 errors)
> druid:qy796rh6653
...snip 1339 druids...
> druid:zn454yx7014
[deploy@sdr-deploy dor-services-app]$ 
```

However, it also seemed to introduce 75 instances of `Contributor type incorrectly capitalized`?  i know contributor mappings have been under active development, so maybe this is drift between `master` (as of Fri evening when I ran the validations) and this PR's base branch?  I will rebase and re-run the validation on this branch.

## Which documentation and/or configurations were updated?

worked off of this updated example:  https://github.com/sul-dlss-labs/cocina-descriptive-metadata/pull/127/files

### CODE REVIEW

**Comment by jmartin-sul** [comment]
results as of a friday evening rebase on `master`:

```
[deploy@sdr-deploy dor-services-app]$ diff results_to-cocina_2020-11-21T05-06-37Z_master_adbb5dc.txt results_to-cocina_2020-11-21T09-43-24Z_dateOther-event-type_e72fa2f.txt
2c2
< Data error: 118268 of 2685579 (4.403817575278925%)
---
> Data error: 114325 of 2685579 (4.256996349762938%)
93255,102768d93254
< Data error: [DATA ERROR] originInfo/dateOther missing eventType (9513 errors)
< druid:xv158sd4671
< druid:qx562pf7510
...
< druid:ds794zt2803
< druid:hk075hj0625
109790a100277,105772
> Data error: [DATA ERROR] originInfo/dateOther missing eventType (5495 errors)
> druid:qy796rh6653
> druid:dm508mt8359
...
> druid:yb649wq0442
> druid:hk075hj0625
113325c109307
< Data error: [DATA ERROR] Contributor type incorrectly capitalized (2667 errors)
---
> Data error: [DATA ERROR] Contributor type incorrectly capitalized (2742 errors)
113332a109315
> druid:mh797tm0200
113342a109326
> druid:rr015xw2692
...
> druid:wm497vd0963
115976a112033
> druid:qh821qx7318
[deploy@sdr-deploy dor-services-app]$ 
```

4018 fewer `originInfo/dateOther missing eventType`, but 75 more `Contributor type incorrectly capitalized`?

**Comment by jmartin-sul** [comment]
not sure this is ready to merge, but does seem ready for review 🙂

File: app/services/cocina/from_fedora/descriptive/event.rb [hunk 1]
```diff
@@ -191,11 +191,24 @@ def build_date(event_type, node)
...
+                  type: node['type'] ? 'date type' : 'calendar'
+                }
+              ]
+            end
             date[:type] = node['point'] if node['point']
           end
         end
 
+        def date_other_event_type(origin)
+          return 'creation' if origin.attr('eventType') == 'production'
```
Comment (justinlittman) [inline app/services/cocina/from_fedora/descriptive/event.rb:207]: To be consistent with other code, would recommend `origin[:eventType]` or `origin['eventType']`.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comment.

File: app/services/cocina/from_fedora/descriptive/event.rb (hunk 1 above)
Comment (jmartin-sul) [reply app/services/cocina/from_fedora/descriptive/event.rb]: fixed

**Pull request merged**